Dev into testnet 2409

beqaabu opened the pull request:
# Description of proposed changes

<describe what this PR is about and why we want it>

---

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

beqaabu committed:
CSUB-1259 add mainnet shorthand (#452)

* add mainnet shorthand

* chore: add empty mainnet spec file for node to compile

* chore: fmt

rustlang-dev committed:
update dependency of polkadot and frontier

rustlang-dev committed:
latest polkadot and frontier.

rustlang-dev committed:
creditcoin runtime polkadot/frontier dependency finished.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rustlang-dev committed:
update frontier rpc trace to the latest version of polkadot.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rustlang-dev committed:
fix overrides evm cache task.

rustlang-dev committed:
fix spawn_frontier_tasks compilation.

rustlang-dev committed:
fix overrides, metrics.

rustlang-dev committed:
fix frontier_backend

rustlang-dev committed:
FrontierBackend

rustlang-dev committed:
fix FrontierBackend

rustlang-dev committed:
fix genesis.

rustlang-dev committed:
fix ChainSpec

rustlang-dev committed:
upgrade cc3 node to latest frontier, polkadot sdk but still some todo

rustlang-dev committed:
add todo!

rustlang-dev committed:
fix and enable offchain-workers-runner spawn.

rustlang-dev committed:
remove 2 todos

rustlang-dev committed:
fix 2 todos about grandpa notification inside new_full node fn.

rustlang-dev committed:
fix run of Subcommand::FrontierDb

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rustlang-dev committed:
fix new_full fn that now match &frontier_backend into EthDeps struct

rustlang-dev committed:
fix spawn_tracing_tasks fn to match *params.frontier_backend.clone()

rustlang-dev committed:
udate readme.

rustlang-dev committed:
fix genesis build. made compatible with stable0724 version. json genesis

rustlang-dev committed:
frontier backend no locks available fix. read description

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rustlang-dev committed:
refactoring after migration to latest polkadot-sdk/frontier.

rustlang-dev committed:
register metrics with prometheus for node service new_full fn

rustlang-dev committed:
cargo fmt --

rustlang-dev committed:
fix some clippy.

rustlang-dev committed:
fix unit tests build. tests passed.

rustlang-dev committed:
remove frontier and polkadot-sdk from check for used forks .sh

rustlang-dev committed:
whitelist github.com/paritytech frontier and polkadot-sdk

rustlang-dev committed:
fix clippy

rustlang-dev committed:
fix all lints to allow deprecated fn

rustlang-dev committed:
cargo fmt

rustlang-dev committed:
check migration locally. new runtime on top of old node + runtime

rustlang-dev committed:
add file to run chopstics

rustlang-dev committed:
update readme about chopsticks

DylanVerstraete committed:
chore: use rust 1.81

beqaabu committed:
chore: fix after rebase

beqaabu committed:
feat: move to polkadot 2409 and moonbeam fork for tracing

beqaabu committed:
chore: fix clippy

beqaabu committed:
fix: add missing host functions and bump version

beqaabu committed:
chore: fix tracing integration test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

beqaabu committed:
feat: experimenting with srtool

beqaabu committed:
fix: make cli work

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

beqaabu committed:
chore: fix js fmt

atodorov committed:
Relax assertions for unbonding interval when queried almost each block

b/c an actual iteration of this for loop may span between 2 consequtive
blocks, or right around the edges of block time!

beqaabu committed:
chore: fix precompiles check and test

beqaabu committed:
chore: add node-key to docker and add timeouts to withdraw-unbonded tests

beqaabu committed:
fix: setup nodes correctly and update extrinsics inside cli

beqaabu committed:
chore: use updated fork for tracing hostfunctions

beqaabu committed:
chore: pr comments

beqaabu committed:
fix: gitleaks for new chopsticks config file

beqaabu committed:
chore: pr comments, docs

beqaabu committed:
chore: gitleaks again

dependabot[bot] committed:
build(deps-dev): bump vite

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.0 to 6.2.4.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.4/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.4/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump the all-dependencies group with 2 updates

Bumps the all-dependencies group with 2 updates: [taiki-e/install-action](https://github.com/taiki-e/install-action) and [oxsecurity/megalinter](https://github.com/oxsecurity/megalinter).


Updates `taiki-e/install-action` from 2.49.9 to 2.49.40
- [Release notes](https://github.com/taiki-e/install-action/releases)
- [Changelog](https://github.com/taiki-e/install-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/taiki-e/install-action/compare/v2.49.9...v2.49.40)

Updates `oxsecurity/megalinter` from 8.3.0 to 8.5.0
- [Release notes](https://github.com/oxsecurity/megalinter/releases)
- [Changelog](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md)
- [Commits](https://github.com/oxsecurity/megalinter/compare/v8.3.0...v8.5.0)

---
updated-dependencies:
- dependency-name: taiki-e/install-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: all-dependencies
- dependency-name: oxsecurity/megalinter
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: all-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump the dev-dependencies group across 1 directory with 4 updates

Bumps the dev-dependencies group with 4 updates in the /cli directory: [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin), [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser), [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) and [ts-jest](https://github.com/kulshekhar/ts-jest).


Updates `@typescript-eslint/eslint-plugin` from 8.25.0 to 8.29.0
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.29.0/packages/eslint-plugin)

Updates `@typescript-eslint/parser` from 8.25.0 to 8.29.0
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.29.0/packages/parser)

Updates `eslint-config-prettier` from 10.0.1 to 10.1.1
- [Release notes](https://github.com/prettier/eslint-config-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v10.0.1...v10.1.1)

Updates `ts-jest` from 29.2.4 to 29.3.1
- [Release notes](https://github.com/kulshekhar/ts-jest/releases)
- [Changelog](https://github.com/kulshekhar/ts-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/kulshekhar/ts-jest/compare/v29.2.4...v29.3.1)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-config-prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: ts-jest
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump openssl from 0.10.70 to 0.10.72

Bumps [openssl](https://github.com/sfackler/rust-openssl) from 0.10.70 to 0.10.72.
- [Release notes](https://github.com/sfackler/rust-openssl/releases)
- [Commits](https://github.com/sfackler/rust-openssl/compare/openssl-v0.10.70...openssl-v0.10.72)

---
updated-dependencies:
- dependency-name: openssl
  dependency-version: 0.10.72
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump tokio from 1.42.0 to 1.44.2

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.42.0 to 1.44.2.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.42.0...tokio-1.44.2)

---
updated-dependencies:
- dependency-name: tokio
  dependency-version: 1.44.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

beqaabu committed:
chore: update version to 3.43 and fix gitleaks

Commit:
Auto-update TypeScript type definitions

beqaabu committed:
chore: update cargo lock

BradleyOlson64 reviewed (dismissed):
Looks good!

atodorov reviewed (changes requested):
Runtime upgrade against a forked chain failed with 
```
Error: NetworkKeyNotFound("/mnt/fork-data/chains/creditcoin3_testnet-fork/network/secret_ed25519")
```

See https://github.com/gluwa/creditcoin3/actions/runs/14645165439/job/41100249041?pr=590

atodorov committed:
Add the --unsafe-force-node-key-generation option when starting a forked chain

this works around an upstream change, see:
https://github.com/paritytech/polkadot-sdk/pull/3852

https://gluwa.slack.com/archives/C03MQ532BGA/p1745508203610989

atodorov commented:
Failure in test-against-fork when running the @gluwa/evm-network-test tool:

```
  1) Validate Standard Contract Transactions
       "before all" hook in "Validate Standard Contract Transactions":
     ProviderError: gas price less than block base fee
```
see https://github.com/gluwa/creditcoin3/actions/runs/14660866844/job/41147418157?pr=590


This is the same as in integration-test-smart-contract, https://github.com/gluwa/creditcoin3/actions/runs/14660866846/job/41145545977?pr=590 . 

This error is unrelated to the polkadot version upgrade. Happened 2 weeks ago for b9287df9e67a9bb1469abe196c35e310d11b2e5a, see here https://github.com/gluwa/creditcoin3/actions/runs/14328217528/job/40159453218

beqaabu committed:
feat: add metadata-hash

beqaabu commented:
this is pr I'm using for dryrun, needs to be open

beqaabu commented:
note to self: wasm build needs to be updated the include `--features metadata-hash` on all branches

atodorov commented:
> note to self: wasm build needs to be updated the include `--features metadata-hash` on all branches

wasn't that going to be enabled by default? We can discuss it later.

beqaabu commented:
> > note to self: wasm build needs to be updated the include `--features metadata-hash` on all branches
> 
> wasn't that going to be enabled by default? We can discuss it later.

yes the initial idea was to have it enabled by default, but upon further reading feature flags make more sense, see https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/enable_metadata_hash/index.html#enable-metadata-hash-generation
>Note:
The metadata-hash feature needs to be enabled for the substrate-wasm-builder to enable the code for being able to generate the metadata hash. It is also recommended to put the metadata hash generation behind a feature in the runtime as shown above. The reason behind is that it adds a lot of code which increases the compile time and the generation itself also increases the compile time. Thus, it is recommended to enable the feature only when the metadata hash is required (e.g. for an on-chain build).

beqaabu commented:
closing in favor of #599
